### DISCUSSION TITLE
"Unless showBlank" message not working in V4.3.0. Was working in v3.1.6.

### DISCUSSION THREAD

**Discussion by annetimdegraaf** [microsoft-search/pnp-modern-search#1416] [Q&A]
**Version used**
4.3.0

**Question**
I'm migrating from the latest v3 to v4.3.0. Everything works great except one thing. When I'm trying to change the "no result" text it won't work. I also checked the default result templates but they also not facilitate the option to show a message when there is no result based on the searchterm. I use the following code in my template:

    {{else}}
        {{#unless showBlank}}
            <div class="template_noResults">TEXT HERE</div>
        {{/unless}}
    {{/if}}`

I think it has something to do with the following line that is not working/available in the default/custom template:

`{{#if @root.hasPrimaryOrSecondaryResults}}`

When you add the unless lines it won't show results anymore.

**Browser**
Any browser

**Comment by wobba** [comment]
You should be able to check for `@root.data.items.length > 0` and `@root.data.secondaryResults.length > 0`

**Comment by annetimdegraaf** [comment]
@wobba thanks for helping. Tried to add this but still does not work. Where should I add it? Is the following correct:

```
{{#if @root.data.items.length > 0}}
                <div>Custom text</div>
        {{/if}}
```

**Comment by wobba** [comment]
@annetimdegraaf check handlebars docs at https://github.com/helpers/handlebars-helpers#array and https://handlebarsjs.com/guide/builtin-helpers.html

Something like below might work - not tested.

```
{{#unless (equalsLength @root.data.items.length 0)}}
...
{{unless}}

**Comment by FranckyC** [comment]
> 
> 
> @wobba thanks for helping. Tried to add this but still does not work. Where should I add it? Is the following correct:
> 
> ```
> {{#if @root.data.items.length > 0}}
>                 <div>Custom text</div>
>         {{/if}}
> ```

Use [`{{#gt}}`](https://aequos-solutions.github.io/modern-data-visualizer/usage/data-visualizer/tokens/) operator instead as this syntax in not a valid Handlebars syntax.

Regarding the initial qeuestion, the `showBlank` property must be accessible whithin the context where your condition is. To acces proeprties from the root context, use `@root.properties.showBlankIfNoResult` instead.
